lib/inferno/dsl/fhir_evaluation/rules/all_search_parameters_have_examples.rb

```diff
@@ -0,0 +1,93 @@
+# frozen_string_literal: true
+
+require_relative '../reference_extractor'
+
+module Inferno
+  module DSL
+    module FHIREvaluation
+      module Rules
+        class AllSearchParametersHaveExamples < Rule
+          attr_accessor :unused_resource_urls
+
+          def check(context)
+            unless ENV.fetch('FHIRPATH_URL')
+              message = 'FHIRPATH_URL is not found. Skipping rule AllSearchParametersHaveExamples.'
+              result = EvaluationResult.new(message, severity: 'warning', rule: self)
+              context.add_result result
+              return
+            end
+
+            @unused_resource_urls = []
+            search_params = context.ig.to_hash[:resources_by_type]['SearchParameter']
+
+            get_unused_resource_urls(context.ig.to_hash[:resources_by_type]['SearchParameter']) do |param|
+              param_is_used?(param, context)
+            end
+
+            if unused_resource_urls.any?
+              message = "Found SearchParameters with no searchable data: \n\t#{unused_resource_urls.join(' ,')}"
+              result = EvaluationResult.new(message, rule: self)
+            elsif !search_params.empty?
+              message = 'All SearchParameters have examples'
+              result = EvaluationResult.new(message, severity: 'success', rule: self)
+            else
+              message = 'IG contains no SearchParameter'
+              result = EvaluationResult.new(message, severity: 'information', rule: self)
+            end
+
+            context.add_result result
+          end
+
+          def get_unused_resource_urls(ig_data, &resource_filter)
```

Comment:
This method doesn't really seem to do anything. Why does it need to take a block?